keps/1618-optional-gc-of-workloads/README.md

```diff
@@ -0,0 +1,440 @@
+was found that removes the resource-in-use 
+finalizer when the Workload state transitions to finished. If that behavior ever 
+changes, or if there is an external source of a finalizer being attached 
+to the Workload, it will be marked for deletion by the Kubernetes client 
+but will not actually be deleted until the finalizer is removed.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+- **R**: In clusters with a large number of existing finished Workloads 
+(thousands, tens of thousands, etc.), it may take a significant amount of time
+to delete all previously finished Workloads that qualify for deletion. Since the 
+reconciliation loop is effectively synchronous, this may impact the reconciliation 
+of new jobs. From the user's perspective, it may seem like kueue's initialization 
+time is very long for that initial run until all expired objects are deleted.\
+**M**: Unfortunately, there is no easy way to mitigate this without complicating the code.
+A potential improvement would be to have a dedicated internal queue that handles deletion 
+outside the reconciliation loop, or to delete expired objects in batches so that after 
+a batch of deletions is completed, any remaining expired objects would be skipped until 
+the next run of the loop. However, this would go against the synchronous nature of the reconciliation loop.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+
+### API
+```go
+// Configuration is the Schema for the kueueconfigurations API
+type Configuration struct {
+    //...
+    // ObjectRetentionPolicies provides configuration options for retention of kueue owned
+    // resources.
+    ObjectRetentionPolicies *ObjectRetentionPolicies `json:"objectRetentionPolicies,omitempty"`
+}
+
+//...
+
+type ObjectRetentionPolicies struct {
+    // FinishedWorkloadRetention is the duration to retain finished Workloads.
+    // A duration of 0 will delete finished Workloads immediately.
+    // A nil value will disable automatic deletion.
+    // The value is represented using the metav1.Duration format, allowing for flexible
+    // specification of time units (e.g., "24h", "1h30m", "30s").
+    //
+    // Defaults to null.
+    // +optional
+    FinishedWorkloadRetention *metav1.Duration `json:"finishedWorkloadRetention"`
```

Comment:
Could this name suggest the field is related to time e.g. `FinishedWorkloadRetentionTime`

Reply:
It could and if you feel strongly about it I'm happy to do it. Personally I think it'd a bit excessive and redundant since retention in this context could only be associated with time. But as I said, if you think it'd improve the clarity of the code and if in your opinion the association isn't obvious and could be mistaken with something else I'll make the change. WDYT?